SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -801,4 +807,109 @@ void FLocalDeploymentManager::SetAutoDeploy(bool bInAutoDeploy)
+			if (!bSucceeded)
+			{
+				UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed to trigger snapshot at '%s'; received '%s'"),
+					   *HttpRequest->GetURL(), *HttpResponse->GetContentAsString());
+				if (OnSnapshotTaken != nullptr)
+				{
+					OnSnapshotTaken(false, FString());
+				}
+				return;
+			}
+
+			// Unfortunately by the time this callback happens, the files haven't been flushed, so if you copy you may get
+			// the wrong info! So let's wait a bit..
+			FTimerHandle TimerHandle;
+			World->GetTimerManager().SetTimer(
+				TimerHandle,
+				[bUseStandard, OnSnapshotTaken]() {
+					bool bSuccess = false;
+
+					FString NewestSnapshotFilePath;
+					FString SnapshotSearchDirectory;
+
+					if (bUseStandard)
+					{
+						FSpatialGDKServicesModule& GDKServices =
+							FModuleManager::GetModuleChecked<FSpatialGDKServicesModule>("SpatialGDKServices");
+						FLocalDeploymentManager* LocalDeploymentManager = GDKServices.GetLocalDeploymentManager();
+
+						FString SpatialPath = FPaths::ProjectDir() + TEXT("../spatial/");
+						SnapshotSearchDirectory =
+							SpatialPath + TEXT("logs/localdeployment/") + LocalDeploymentManager->GetLocalRunningDeploymentID();
+
+						IFileManager& FileManager = FFileManagerGeneric::Get();
+
+						TArray<FString> SnapshotFiles;
+
+						FileManager.FindFiles(SnapshotFiles, *SnapshotSearchDirectory, TEXT("snapshot"));
+
+						FDateTime NewestSnapshotTimestamp = FDateTime::MinValue();
+
+						for (const FString& SnapshotFile : SnapshotFiles)
+						{
+							FString SnapshotFilePath = SnapshotSearchDirectory + TEXT("/") + SnapshotFile;
+							FDateTime SnapshotFileTimestamp = FileManager.GetTimeStamp(*SnapshotFilePath);
+							if (SnapshotFileTimestamp > NewestSnapshotTimestamp)
+							{
+								NewestSnapshotTimestamp = SnapshotFileTimestamp;
+								NewestSnapshotFilePath = SnapshotFilePath;
+							}
+						}
+					}
+					else
+					{
+						FString AppDataLocalPath = FPlatformMisc::GetEnvironmentVariable(TEXT("LOCALAPPDATA"));
+						SnapshotSearchDirectory = FString::Printf(TEXT("%s/.improbable/local_snapshots"), *AppDataLocalPath);
+						FString LatestSnapshotInfoPath = FString::Printf(TEXT("%s/latest"), *SnapshotSearchDirectory);
+						FString LatestSnapshot;
+						if (FPaths::FileExists(LatestSnapshotInfoPath)
+							&& FFileHelper::LoadFileToString(LatestSnapshot, *LatestSnapshotInfoPath))
+						{
+							NewestSnapshotFilePath =
+								FString::Printf(TEXT("%s/.improbable/local_snapshots/%s"), *AppDataLocalPath, *LatestSnapshot);
+						}
+					}
+
+					bSuccess = !NewestSnapshotFilePath.IsEmpty();
+
+					if (!bSuccess)
+					{
+						UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed find snapshot file in '%s'"), *SnapshotSearchDirectory);
+					}
+
+					if (OnSnapshotTaken != nullptr)
+					{
+						OnSnapshotTaken(bSuccess, NewestSnapshotFilePath);
+					}
+				},
+				0.5f, false);
```

Comment:
Same here regarding the /* InVarName */ for undeclared vars.

Reply:
In the case that the snapshot has not yet been flushed before 0.5s shouldn't we be doing an expotential backoff and a few retries? Just for some more stable code depending on machine load.

Reply:
There's a task to fix this in order to have it only respond after the snapshot has been taken and saved, this is just a temporary thing for now https://improbableio.atlassian.net/browse/SQD-871